Source/CombatExtended/Compatibility/BlockerRegistry.cs

```diff
@@ -88,20 +92,19 @@ public static void RegisterBeforeCollideWithCallback(Func<ProjectileCE, Thing, b
             }
             beforeCollideWithCallbacks.Add(f);
         }
-        public static bool CheckForCollisionBetweenCallback(ProjectileCE projectile, Vector3 from, Vector3 to)
+        public static IEnumerable<(Vector3 IntersectionPos, Action OnInterception)> CheckForCollisionBetweenCallback(ProjectileCE projectile, Vector3 from, Vector3 to)
```

Comment:
As mentioned above, an in-line filter / accumulator pattern will have substantially less overhead.  Not much we can do about the inner generators though.  

We could require each callback to do its own filtering, but odds are good new ones will fall back into returning the first shield rather than the closest shield, and it would end up duplicating code.

```
float curDistance = float.MaxValue;
(Vector3, Action)? curLocation = null;
foreach ...
{
    foreach ...
    {
        if (distance < curDistance)
        {
            curDistance = distance;
            curLocation = location;
        }
    }
}
return curLocation;
```

Reply:
Done